<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="pioneer3dx" >

  <!-- Import Rviz colors -->
  <xacro:include filename="$(find pioneer3dx)/xacro/materials.xacro" />

  <!-- Import robot -->
  <xacro:include filename="$(find pioneer3dx)/xacro/p3dx/pioneer3dx_body.xacro"/>
  <xacro:pioneer3dx_body name="chassis" parent="base_link" meshes="package://pioneer3dx/meshes/p3dx"/>

  <!-- Laser  -->
  <xacro:include filename="$(find pioneer3dx)/xacro/laser/hokuyo.xacro"/>
  <xacro:laser name="front_laser" parent="chassis" xyz="0.16 0 0.1" rpy="0 0 0" meshes="package://pioneer3dx/meshes/laser"/>


  <!-- GPU Laser
  <xacro:include filename="$(find pioneer3dx)/xacro/laser/hokuyo_gpu.xacro"/>
  <xacro:laser_gpu name="front_laser" parent="chassis" xyz="0.16 0 0.1" rpy="0 0 0" meshes="package://pioneer3dx/meshes/laser" />
    -->
  <xacro:include filename="$(find pioneer3dx)/xacro/camera/camera.xacro"/>
  <xacro:cam name="front_camera" parent="chassis" xyz="0 0 0.5" rpy="0 0 0" />

</robot>
